benchmarks/models.py

```diff
@@ -0,0 +1,77 @@
+import math
+
+import torch
+import torch.nn as nn
+
+
+class EmbeddingLayer(nn.Embedding):
+    def __init__(self, ntoken, ninp, initrange):
+        super().__init__(ntoken, ninp)
+        self.ninp = ninp
+        self.weight.data.uniform_(-initrange, initrange)
+
+    def forward(self, src):
+        return super().forward(src) * math.sqrt(self.ninp)
+
+
+class PositionalEncodingLayer(nn.Module):
```

Comment:
is there a pytorch lib we can use or we need to write our own positional embedding?

Reply:
ignore that. I didn't find standard lib for that. perhaps just add a docstring here like:

""" This implements the positional encoding for original Transformer paper. """

Reply:
We can use the [nn.transformer](https://pytorch.org/docs/master/nn.html#transformer-layers) which provides the entire model or specific encoder/decoder layers. There are no specific positional encoding layers provided. The [pytorch tutorial] (https://pytorch.org/tutorials/beginner/transformer_tutorial.html) seemed to have it's own PositionalEncodingLayer as well.

Reply:
OK to leave it in as our own implementation. It is not very complex anyway. Good comments will be nice to have.